modin/core/dataframe/pandas/dataframe/utils.py

```diff
@@ -17,287 +17,416 @@
+        to use to roughly sort the entire dataframe. It does so by collating all the samples
+        and computing NPartitions.get() quantiles for the overall set.
+
+        Parameters
+        ----------
+        samples : np.ndarray
+            The samples computed by ``get_partition_quantiles_for_sort``.
+        ideal_num_new_partitions : int
+            The ideal number of new partitions.
+        method : str, default: linear
+            The method to use when picking quantiles.
+        key : Callable, default: None
+            The key to use on the samples when picking pivots.
+
+        Returns
+        -------
+        np.ndarray
+            A list of overall quantiles.
+        """
+        # We don't call `np.unique` on the samples, since if a quantile shows up in multiple
+        # partition's samples, this is probably an indicator of skew in the dataset, and we
+        # want our final partitions to take this into account.
+        if key is not None:
+            samples = key(samples)
+        # We don't want to pick very many quantiles if we have a very small dataframe.
+        num_quantiles = ideal_num_new_partitions
+        quantiles = [i / num_quantiles for i in range(1, num_quantiles)]
+        # If we only desire 1 partition, we need to ensure that we're not trying to find quantiles
+        # from an empty list of pivots.
+        if len(quantiles) > 0:
+            return cls._find_quantiles(samples, quantiles, method)
+        return np.array([])
+
+    @staticmethod
+    def split_partitions_using_pivots_for_sort(
+        df: pandas.DataFrame,
+        columns_info: "list[ColumnInfo]",
+        ascending: bool,
+        **kwargs: dict,
+    ) -> "tuple[pandas.DataFrame, ...]":
+        """
+        Split the given dataframe into the partitions specified by `pivots` in `columns_info`.
+
+        This function takes as input a row-axis partition, as well as the quantiles determined
+        by the `pivot_func` defined above. It then splits the input dataframe into NPartitions.get()
+        dataframes, with the elements in the i-th split belonging to the i-th partition, as determined
+        by the quantiles we're using.
+
+        Parameters
+        ----------
+        df : pandas.Dataframe
+            The partition to split.
+        columns_info : list of ColumnInfo
+            Information regarding keys and pivots for range partitioning.
+        ascending : bool
+            The ascending flag.
+        **kwargs : dict
+            Additional keyword arguments.
+
+        Returns
+        -------
+        tuple[pandas.DataFrame]
+            A tuple of the splits from this partition.
+        """
+        if len(columns_info) == 0:
+            # We can return the dataframe with zero changes if there were no pivots passed
+            return (df,)
+
+        na_index = (
+            df[[col_info.name for col_info in columns_info]].isna().squeeze(axis=1)
+        )
+        if na_index.ndim == 2:
```

Comment:
`ndim == 2` in case we had taken multiple columns here
```python
>>> df[["a"]].squeeze(axis=1).ndim
1
>>> df[["a", "b"]].squeeze(axis=1).ndim
2
```